components/eventing-controller/pkg/object/equality.go

```diff
@@ -48,3 +58,253 @@ func apiRuleEqual(a1, a2 *apigatewayv1alpha1.APIRule) bool {
+		return false
+	}
+
+	return true
+}
+
+// containerEqual asserts the equality of two Container objects.
+func containerEqual(c1, c2 *corev1.Container) bool {
+	if c1 == nil || c2 == nil {
+		return false
+	}
+	if c1.Image != c2.Image {
+		return false
+	}
+
+	ps1, ps2 := c1.Ports, c2.Ports
+	if len(ps1) != len(ps2) {
+		return false
+	}
+	for i := range ps1 {
+		p1, p2 := &ps1[i], &ps2[i]
+
+		if p1.Name != p2.Name ||
+			p1.ContainerPort != p2.ContainerPort ||
+			realProto(p1.Protocol) != realProto(p2.Protocol) {
+
+			return false
+		}
+	}
+
+	if !envEqual(c1.Env, c2.Env) {
+		return false
+	}
+
+	if !probeEqual(c1.ReadinessProbe, c2.ReadinessProbe) {
+		return false
+	}
+
+	return true
+}
+
+func envEqual(e1, e2 []corev1.EnvVar) bool {
+	if len(e1) != len(e2) {
+		return false
+	}
+EV1:
+	for _, ev1 := range e1 {
+		for _, ev2 := range e2 {
+			if reflect.DeepEqual(ev1, ev2) {
+				continue EV1
+			}
+		}
+		return false
+	}
+	return true
+}
+
+// probeEqual asserts the equality of two Probe objects.
+func probeEqual(p1, p2 *corev1.Probe) bool {
+	if p1 == nil || p2 == nil {
+		return false
+	}
+
+	if p1 == p2 {
+		return true
+	}
+
+	isInitialDelaySecondsEqual := p1.InitialDelaySeconds != p2.InitialDelaySeconds
+	isTimeoutSecondsEqual := p1.TimeoutSeconds != p2.TimeoutSeconds && p1.TimeoutSeconds != 0 && p2.TimeoutSeconds != 0
+	isPeriodSecondsEqual := p1.PeriodSeconds != p2.PeriodSeconds && p1.PeriodSeconds != 0 && p2.PeriodSeconds != 0
+	// Knative sets a default when that value is 0
+	isSuccessThresholdEqual := p1.SuccessThreshold != p2.SuccessThreshold && p1.SuccessThreshold != 0 && p2.SuccessThreshold != 0
+	isFailureThresholdEqual := p1.FailureThreshold != p2.FailureThreshold && p1.FailureThreshold != 0 && p2.FailureThreshold != 0
+
+	if isInitialDelaySecondsEqual || isTimeoutSecondsEqual || isPeriodSecondsEqual || isSuccessThresholdEqual || isFailureThresholdEqual {
+		return false
+	}
+
+	if !handlerEqual(&p1.Handler, &p2.Handler) {
+		return false
+	}
+
+	return true
+}
+
+// handlerEqual asserts the equality of two Handler objects.
+func handlerEqual(h1, h2 *corev1.Handler) bool {
+	if h1 == h2 {
+		return true
+	}
+	if h1 == nil || h2 == nil {
+		return false
+	}
+
+	hg1, hg2 := h1.HTTPGet, h2.HTTPGet
+	if hg1 == nil && hg2 != nil {
+		return false
+	}
+	if hg1 != nil {
+		if hg2 == nil {
+			return false
+		}
+
+		if hg1.Path != hg2.Path {
+			return false
+		}
+	}
+
+	return true
+}
+
+// default Protocol is TCP, so we assume empty equals TCP
+// https://godoc.org/k8s.io/api/core/v1#ServicePort
+func realProto(pr corev1.Protocol) corev1.Protocol {
+	if pr == "" {
+		return corev1.ProtocolTCP
+	}
+	return pr
+}
+
+// secretEqual asserts the equality of two Secret objects for event publisher proxy deployments.
+func secretEqual(b1, b2 *corev1.Secret) bool {
+	if b1 == nil || b2 == nil {
+		return false
+	}
+
+	if b1 == b2 {
+		return true
+	}
+
+	if b1 == nil || b2 == nil {
```

Comment:
Already checked first thing in the functiuon.

Reply:
True, removed